Several Updates on  the BP document

bernafarias opened the pull request.

newtoncalegari commented:
Closing this pull request because Bernadette will make a new one with BP document updated.

bernafarias committed:
Updated several sections

- Introduction section updated
- Audience section updated
- Data Vocabularies section updated
- Inclusion of new BP based on the requirements already identified in
the Use Cases document
- Inclusion of issues with comments and questions to be discussed with
the group
